docs/providers/okta.md

```diff
@@ -0,0 +1,174 @@
+# Okta Provider
+
+[Source Code](https://github.com/nuxt-community/auth-module/blob/dev/lib/providers/okta.js)
+
+## IMPORTANT NOTES:
+
+1. This Okta Provider is designed to use the ["implicit flow"](https://oauth.net/2/grant-types/implicit/) and to always redirect to Okta.com for login and logout.
+
+2. Okta defaults to `/implicit/callback` instead of `callback`, so you _MUST_ ensure that you _EITHER_ have a corresponding callback component (i.e. `pages/implicit/callback.vue`) _OR_ that you change your Okta settings (Applications >> \<app\> >> General >> Logout redirect URIs).
+
+3. These instructions use the `@nuxtjs/dotenv` module extensively.  This module and approach isn't strictly necessary, but is *_highly_* recommended, particularly if you are going to be using different Okta domains or server ids across environments (e.g. for different stages of CI).
```

Comment:
I think _highly_ recommended should have a reason and IMO it is not recommended to use a `.env` file, especially for CI/CD, flows but recommend using environment variables. Suggest to read [The Twelve-Factor App](https://12factor.net/)

Reply:
imo its useful for local dev it keeps you env variables isolated per project and someone can transfer their `.env` file for quick dev setup and it is easy to add or remove env variables.

Reply:
@noobling I would be happy to discuss more but I think is out of the scope of this PR review and even if we finally end up recommended `.env` file, it should be module-wise not just for a specific provider that with the implicit flow doesn't even contain a server secret.

Reply:
@pi0 I agree it is out of scope of this PR and should be removed.

From my understanding using a `.env` file _is_ using environment variables. https://cli.vuejs.org/guide/mode-and-env.html#environment-variables. 

Using separate `.env` files per environment/mode and adding them to .gitignore seems to work well foor CI/CD. I'm absolutely open to other methods for configuration of single-page apps. (although out of scope of this PR)